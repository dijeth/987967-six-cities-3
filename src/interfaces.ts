import { id, roomType, coord } from "./types";

export interface OfferMini {
    readonly id: id;
    title: string;
    type: roomType;
    pictures: Array<string>;
    cost: number;
    rating: number;
    isPremium: boolean;
    isFavorite: boolean;
    city: string;
    coord: coord;
};

export interface Offer extends OfferMini {
    bedroomCount: number;
    adultsCount: number;
    insideFeatures: Array<string>;
    userName: string;
    userPicture: string;
    isSuperUser: boolean;
    descriptionTitle: string;
    description: string;
    zoom: number;
}

export interface City {
    name: string;
    zoom: number;
    centerCoord: coord;
}

export interface UserReview {
    readonly id: id;
    userName: string;
    userPicture: string;
    rating: number;
    description: string;
    date: string;
}