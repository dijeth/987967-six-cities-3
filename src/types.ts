import { SortType } from "./const/const";
import { OfferMini, City } from "./interfaces";

export type id = string;
export type roomType = `apartment` | `room` | `house` | `hotel`;
export type coord = [number, number];
export type ListItem = SortType | OfferMini | City;
export type TSavedReview = {
    text: string;
    rating: number;
    offerID: id;
};
