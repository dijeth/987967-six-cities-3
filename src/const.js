const OfferType = {
  APARTMENT: `Apartment`,
  PRIVATE_ROOM: `Private room`
};

const CITIES = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

export {OfferType, CITIES};
