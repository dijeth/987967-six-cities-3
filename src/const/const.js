"use strict";
exports.__esModule = true;
exports.OfferType = {
    APARTMENT: "Apartment",
    PRIVATE_ROOM: "Private room"
};
exports.CITIES = [
    "Paris",
    "Cologne",
    "Brussels",
    "Amsterdam",
    "Hamburg",
    "Dusseldorf"
];
exports.InsideFeature = {
    WI_FI: "Wi-Fi",
    WASHING_MACHINE: "Washing machine",
    TOWELS: "Towels",
    HEATING: "Heating",
    COFFEE_MACHINE: "Coffee machine",
    BABY_SEAT: "Baby seat",
    KITCHEN: "Kitchen",
    DISHWASHER: "Dishwasher",
    CABEL_TV: "Cabel TV",
    FRIDGE: "Fridge"
};
exports.ScreenType = {
    MAIN: "MAIN",
    PROPERTY: "PROPERTY"
};
exports.BREAK_STRING = "\n";
exports.MONTHS = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
exports.MAX_CITY_COUNT = 6;
var SortType;
(function (SortType) {
    SortType["POPULAR"] = "Popular";
    SortType["PRICE_LOW_TO_HIGH"] = "Price: low to high";
    SortType["PRICE_HIGH_TO_LOW"] = "Price: high to low";
    SortType["TOP_RATED_FIRST"] = "Top rated first";
})(SortType = exports.SortType || (exports.SortType = {}));
exports.SORT_LIST = [
    SortType.POPULAR,
    SortType.PRICE_LOW_TO_HIGH,
    SortType.PRICE_HIGH_TO_LOW,
    SortType.TOP_RATED_FIRST
];
exports.AuthorizationStatus = {
    NO_AUTH: "NO_AUTH",
    AUTH: "AUTH"
};
exports.AppRoute = {
    getLogin: function () { return "/login"; },
    getOffer: function (id) { return "/offer/" + id; },
    getFavorites: function () { return "/favorites"; },
    getRoot: function () { return "/"; }
};
exports.ServerRoute = {
    getLogin: function () { return "/login"; },
    getFavorites: function () { return "/favorite"; },
    postFavorites: function (id, status) { return "/favorite/" + id + "/" + status; },
    getNearby: function (id) { return "/hotels/" + id + "/nearby"; },
    getHotels: function () { return "/hotels"; },
    getComments: function (id) { return "/comments/" + id; }
};
exports.MAX_IMAGE_COUNT = 6;
exports.ReviewLength = {
    MIN: 50,
    MAX: 300
};
exports.ServerError = {
    UNAUTHORIZED: 401,
    BAD_REQUEST: 400,
    NOT_FOUND: 404
};
var PlaceCardType;
(function (PlaceCardType) {
    PlaceCardType["DEFAULT"] = "DEFAULT";
    PlaceCardType["FAVORITE"] = "FAVORITE";
    PlaceCardType["NEARBY"] = "NEARBY";
})(PlaceCardType = exports.PlaceCardType || (exports.PlaceCardType = {}));
exports.EMPTY_REVIEW = {
    text: "",
    rating: 0,
    offerID: null
};
exports.BASE_URL = "https://htmlacademy-react-3.appspot.com/six-cities";
exports.TIME_OUT = 5000;
