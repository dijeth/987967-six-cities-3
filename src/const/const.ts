export const OfferType = {
  APARTMENT: `Apartment`,
  PRIVATE_ROOM: `Private room`
};

export const CITIES = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

export const InsideFeature = {
  WI_FI: `Wi-Fi`,
  WASHING_MACHINE: `Washing machine`,
  TOWELS: `Towels`,
  HEATING: `Heating`,
  COFFEE_MACHINE: `Coffee machine`,
  BABY_SEAT: `Baby seat`,
  KITCHEN: `Kitchen`,
  DISHWASHER: `Dishwasher`,
  CABEL_TV: `Cabel TV`,
  FRIDGE: `Fridge`
};

export const ScreenType = {
  MAIN: `MAIN`,
  PROPERTY: `PROPERTY`
};

export const BREAK_STRING = `\n`;

export const MONTHS = [`January`, `February`, `March`, `April`, `May`, `June`, `July`, `August`, `September`, `October`, `November`, `December`];

export const MAX_CITY_COUNT = 6;

export enum SortType {
  POPULAR = `Popular`,
  PRICE_LOW_TO_HIGH = `Price: low to high`,
  PRICE_HIGH_TO_LOW = `Price: high to low`,
  TOP_RATED_FIRST = `Top rated first`,
};

export const SORT_LIST = [
  SortType.POPULAR,
  SortType.PRICE_LOW_TO_HIGH,
  SortType.PRICE_HIGH_TO_LOW,
  SortType.TOP_RATED_FIRST
];

export const AuthorizationStatus = {
  NO_AUTH: `NO_AUTH`,
  AUTH: `AUTH`,
};

export const AppRoute = {
  getLogin: () => `/login`,
  getOffer: (id) => `/offer/${id}`,
  getFavorites: () => `/favorites`,
  getRoot: () => `/`,
};

export const ServerRoute = {
  getLogin: () => `/login`,
  getFavorites: () => `/favorite`,
  postFavorites: (id, status) => `/favorite/${id}/${status}`,
  getNearby: (id) => `/hotels/${id}/nearby`,
  getHotels: () => `/hotels`,
  getComments: (id) => `/comments/${id}`,
};

export const MAX_IMAGE_COUNT = 6;

export const ReviewLength = {
  MIN: 50,
  MAX: 300
};

export const ServerError = {
  UNAUTHORIZED: 401,
  BAD_REQUEST: 400,
  NOT_FOUND: 404
};

export enum PlaceCardType {
  DEFAULT = `DEFAULT`,
  FAVORITE = `FAVORITE`,
  NEARBY = `NEARBY`,
};

export const EMPTY_REVIEW = {
  text: ``,
  rating: 0,
  offerID: null,
};

export const BASE_URL = `https://htmlacademy-react-3.appspot.com/six-cities`;
export const TIME_OUT = 5000;
