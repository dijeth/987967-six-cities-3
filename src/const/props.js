import PropTypes from 'prop-types';

export const offerPropType = PropTypes.shape({
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  pictures: PropTypes.arrayOf(PropTypes.string).isRequired,
  cost: PropTypes.number.isRequired,
  rating: PropTypes.number.isRequired,
  isPremium: PropTypes.bool.isRequired,
  isFavorite: PropTypes.bool.isRequired,
  city: PropTypes.string.isRequired,
  coord: PropTypes.arrayOf(PropTypes.number).isRequired
});

export const cityPropType = PropTypes.shape({
  name: PropTypes.string.isRequired,
  zoom: PropTypes.number.isRequired,
  centerCoord: PropTypes.arrayOf(PropTypes.number)
});
