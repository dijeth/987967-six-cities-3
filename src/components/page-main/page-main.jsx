import React from 'react';
import PropTypes from 'prop-types';
import PlaceCardList from '../place-card-list/place-card-list.jsx';
import {SORT_LIST, MainScreenType} from '../../const/const.js';
import {cityPropType} from '../../const/props.js';
import OffersMap from '../offers-map/offers-map.jsx';
import CityList from '../city-list/city-list.jsx';
import SortList from '../sort-list/sort-list.jsx';
import Header from '../header/header.jsx';
import MainEmpty from '../main-empty/main-empty.jsx';
import {connect} from 'react-redux';
import {getSortType, getActiveCity} from '../../reducers/app/selectors.js';
import {getCities} from '../../reducers/data/selectors.js';
import {getSortedOffers} from '../../reducers/selectors.js';
import {getAuthorizationStatus} from '../../reducers/user/selectors.js';
import withOpenState from '../../hocs/with-open-state/with-open-state.jsx';
import withLoading from '../../hocs/with-loading/with-loading.jsx';
import SignIn from '../sign-in/sign-in.jsx';

const SortListWithOpenState = withOpenState(SortList);

const PageMain = ({offers, cities, activeCity, sortType}) => {

  if (offers.length === 0 || activeCity === null) {
    return null;
  }

  const {name: cityName, centerCoord, zoom} = activeCity;
  const offersCoord = offers.map((it) => it.coord);
  const placesCount = offers.length;

  return (
    <div className="page page--gray page--main">
      <Header isActiveLogo={true} isAuthorized={false} />
      <main className={`page__main page__main--index ${offers.length === 0 ? `page__main--index-empty` : ``}`}>
        <h1 className="visually-hidden">Cities</h1>
        <div className="tabs">
          <section className="locations container">
            <CityList items={cities} activeItem={activeCity} />
          </section>
        </div>
        <div className="cities">
          <div className="cities__places-container container">
            <section className="cities__places places">
              <h2 className="visually-hidden">Places</h2>
              <b className="places__found">{placesCount} places to stay in {cityName}</b>
              <SortListWithOpenState items={SORT_LIST} activeItem={sortType} />
              <PlaceCardList items={ offers } isNearPlaces={false} />
            </section>
            <div className="cities__right-section">
              <section className='cities__map map'>
              {<OffersMap centerCoord={centerCoord} offersCoord={offersCoord} zoom={zoom} />}
              </section>
            </div>
          </div>
        </div>
      </main>
    </div>
  );
};

PageMain.propTypes = {
  activeCity: cityPropType,
  cities: PropTypes.arrayOf(cityPropType).isRequired,
  offers: PropTypes.array.isRequired,
  sortType: PropTypes.string,
};

const mapStateToProps = (state) => ({
  sortType: getSortType(state),
  activeCity: getActiveCity(state),
  cities: getCities(state),
  offers: getSortedOffers(state),
  isAuthorized: getAuthorizationStatus(state)
});

export {PageMain};
export default withLoading(connect(mapStateToProps)(PageMain));
