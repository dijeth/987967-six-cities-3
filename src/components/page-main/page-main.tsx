import * as React from 'react';
import OffersMain from '../offers-main/offers-main';
import {SORT_LIST, SortType} from '../../const/const';
import {cityPropType} from '../../const/props.js';
import OffersMap from '../offers-map/offers-map.jsx';
import CityList from '../city-list/city-list';
import SortList from '../sort-list/sort-list';
import Header from '../header/header';
import {connect} from 'react-redux';
import {getAuthorizationStatus} from '../../reducers/user/selectors.js';
import {getSortType, getActiveCity} from '../../reducers/app/selectors.js';
import {getCities} from '../../reducers/data/selectors.js';
import {getCityOffersCoords} from '../../reducers/data/reselectors.js';
import withOpenState from '../../hocs/with-open-state/with-open-state.jsx';
import withLoading from '../../hocs/with-loading/with-loading.jsx';
import withPageError from '../../hocs/with-page-error/with-page-error.jsx';
import { City } from '../../interfaces';
import { coord } from '../../types';

const SortListWithOpenState = withOpenState(SortList);

type Props = {
  activeCity: City;
  cities: Array<City>;
  offersCoord: Array<coord>;
  sortType: SortType;
  isAuth: boolean;
};

const PageMain: React.FC<Props> = ({offersCoord, cities, activeCity, sortType, isAuth}) => {

  const {name: cityName, centerCoord, zoom} = activeCity;
  const placesCount = offersCoord.length;

  return (
    <div className="page page--gray page--main">
      <Header isActiveLogo={true} />
      <main className={`page__main page__main--index ${placesCount === 0 ? `page__main--index-empty` : ``}`}>
        <h1 className="visually-hidden">Cities</h1>
        <div className="tabs">
          <section className="locations container">
            <CityList items={cities} activeItem={activeCity} />
          </section>
        </div>
        <div className="cities">
          <div className="cities__places-container container">
            <section className="cities__places places">
              <h2 className="visually-hidden">Places</h2>
              <b className="places__found">{placesCount} places to stay in {cityName}</b>
              <SortListWithOpenState items={SORT_LIST} activeItem={sortType} />
              <OffersMain isAuth={isAuth}/>
            </section>
            <div className="cities__right-section">
              <section className='cities__map map'>
                <OffersMap centerCoord={centerCoord} offersCoord={offersCoord} zoom={zoom} />
              </section>
            </div>
          </div>
        </div>
      </main>
    </div>
  );
};

const mapStateToProps = (state) => ({
  sortType: getSortType(state),
  activeCity: getActiveCity(state),
  cities: getCities(state),
  offersCoord: getCityOffersCoords(state),
  isAuth: getAuthorizationStatus(state)
});

export {PageMain};
export default withPageError(withLoading(connect(mapStateToProps)(PageMain)));
