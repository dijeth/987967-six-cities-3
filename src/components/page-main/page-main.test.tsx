import * as React from 'react';
import * as Enzyme from 'enzyme';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {BrowserRouter} from 'react-router-dom';
import {PageMain} from './page-main';
import {SortType} from '../../const/const';
import NameSpace from '../../reducers/name-space.js';
import { City, OfferMini } from '../../interfaces.js';
import { coord } from '../../types';

const mockStore = configureStore([]);

const cities: Array<City> = [{
  name: `city-1`,
  centerCoord: [1, 2],
  zoom: 3
}];

const offers: Array<OfferMini> = [{
  city: `city-1`,
  pictures: [`picture-1`],
  title: `title-1`,
  isFavorite: true,
  isPremium: false,
  rating: 1,
  type: `apartment`,
  cost: 1,
  coord: [11, 111],
  id: `1`
},
{
  city: `city-1`,
  pictures: [`picture-2`],
  title: `title-2`,
  isFavorite: true,
  isPremium: false,
  rating: 1,
  type: `apartment`,
  cost: 2,
  coord: [22, 222],
  id: `2`
},
{
  city: `city-1`,
  pictures: [`picture-3`],
  title: `title-3`,
  isFavorite: true,
  isPremium: false,
  rating: 1,
  type: `apartment`,
  cost: 3,
  coord: [33, 333],
  id: `3`
},
{
  city: `city-1`,
  pictures: [`picture-4`],
  title: `title-4`,
  isFavorite: true,
  isPremium: false,
  rating: 1,
  type: `apartment`,
  cost: 4,
  coord: [44, 444],
  id: `4`
},
];

const offersCoord: Array<coord> = [
  [11, 111],
  [22, 222],
  [33, 333],
  [44, 444],
];

const store = mockStore({
  [NameSpace.USER]: {
    isAuthorized: false
  },
  [NameSpace.APP]: {
    activeOffer: null
  },
  [NameSpace.DATA]: {
    offers
  }
});

it(`<PageMain /> should be rendered correctly`, () => {
  const div = document.createElement(`div`);
  document.body.appendChild(div);

  const tree = Enzyme.mount(
      <Provider store={store}>
        <BrowserRouter>
          <PageMain
            activeCity={cities[0]}
            cities={cities}
            offersCoord={offersCoord}
            sortType={SortType.POPULAR}
            isAuth={true}
          />
        </BrowserRouter>
      </Provider>, {attachTo: div});

  expect(tree.getDOMNode()).toMatchSnapshot();
});
