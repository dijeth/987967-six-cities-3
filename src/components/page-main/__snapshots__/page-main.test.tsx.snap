// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageMain /> should be rendered correctly 1`] = `
<div
  class="page page--gray page--main"
>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <div
        class="header__wrapper"
      >
        <div
          class="header__left"
        >
          <a
            class="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              class="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          class="header__nav"
        >
          <ul
            class="header__nav-list"
          >
            <li
              class="header__nav-item user"
            >
              <a
                class="header__nav-link header__nav-link--profile"
                href="/login"
              >
                <div
                  class="header__avatar-wrapper  user__avatar-wrapper"
                />
                <span
                  class="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    class="page__main page__main--index "
  >
    <h1
      class="visually-hidden"
    >
      Cities
    </h1>
    <div
      class="tabs"
    >
      <section
        class="locations container"
      >
        <ul
          class="locations__list tabs__list"
        >
          <li
            class="locations__item"
          >
            <a
              class="locations__item-link tabs__item tabs__item--active"
              href="#"
            >
              <span
                data-index="0"
              >
                city-1
              </span>
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      class="cities"
    >
      <div
        class="cities__places-container container"
      >
        <section
          class="cities__places places"
        >
          <h2
            class="visually-hidden"
          >
            Places
          </h2>
          <b
            class="places__found"
          >
            4
             places to stay in 
            city-1
          </b>
          <form
            action="#"
            class="places__sorting"
            method="get"
          >
            <span
              class="places__sorting-caption"
            >
              Sort by
            </span>
            <span
              class="places__sorting-type"
              tabindex="0"
            >
              Popular
              <svg
                class="places__sorting-arrow"
                height="4"
                width="7"
              >
                <use
                  xlink:href="#icon-arrow-select"
                />
              </svg>
            </span>
            <ul
              class="places__options places__options--custom "
            >
              <li
                class="places__option places__option--active"
                tabindex="0"
              >
                Popular
              </li>
              <li
                class="places__option "
                tabindex="0"
              >
                Price: low to high
              </li>
              <li
                class="places__option "
                tabindex="0"
              >
                Price: high to low
              </li>
              <li
                class="places__option "
                tabindex="0"
              >
                Top rated first
              </li>
            </ul>
          </form>
          <div
            class="cities__places-list tabs__content places__list"
          />
        </section>
        <div
          class="cities__right-section"
        >
          <section
            class="cities__map map"
          >
            <div
              class="leaflet-container leaflet-grab leaflet-touch-drag"
              id="map"
              style="width: 100%; height: 100%; position: relative;"
              tabindex="0"
            >
              <div
                class="leaflet-pane leaflet-map-pane"
                style="left: 0px; top: 0px;"
              >
                <div
                  class="leaflet-pane leaflet-tile-pane"
                >
                  <div
                    class="leaflet-layer "
                    style="z-index: 1;"
                  >
                    <div
                      class="leaflet-tile-container leaflet-zoom-animated"
                      style="z-index: 18; left: 0px; top: 0px;"
                    >
                      <img
                        alt=""
                        class="leaflet-tile"
                        role="presentation"
                        src="https://b.basemaps.cartocdn.com/rastertiles/voyager_labels_under/3/4/3{@2x}.png"
                        style="width: 256px; height: 256px; left: -11px; top: -250px;"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="leaflet-pane leaflet-shadow-pane"
                />
                <div
                  class="leaflet-pane leaflet-overlay-pane"
                />
                <div
                  class="leaflet-pane leaflet-marker-pane"
                >
                  <img
                    alt=""
                    class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                    src="img/pin.svg"
                    style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 620px; top: -57px; z-index: -57;"
                    tabindex="0"
                  />
                  <img
                    alt=""
                    class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                    src="img/pin.svg"
                    style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 1252px; top: -122px; z-index: -122;"
                    tabindex="0"
                  />
                  <img
                    alt=""
                    class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                    src="img/pin.svg"
                    style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 1883px; top: -193px; z-index: -193;"
                    tabindex="0"
                  />
                  <img
                    alt=""
                    class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                    src="img/pin.svg"
                    style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 2515px; top: -273px; z-index: -273;"
                    tabindex="0"
                  />
                </div>
                <div
                  class="leaflet-pane leaflet-tooltip-pane"
                />
                <div
                  class="leaflet-pane leaflet-popup-pane"
                />
              </div>
              <div
                class="leaflet-control-container"
              >
                <div
                  class="leaflet-top leaflet-left"
                />
                <div
                  class="leaflet-top leaflet-right"
                />
                <div
                  class="leaflet-bottom leaflet-left"
                />
                <div
                  class="leaflet-bottom leaflet-right"
                >
                  <div
                    class="leaflet-control-attribution leaflet-control"
                  >
                    <a
                      href="https://leafletjs.com"
                      title="A JS library for interactive maps"
                    >
                      Leaflet
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;
